godot/src/decentraland_components/video_player.gd

```diff
@@ -1,28 +1,531 @@
+		current_backend = BackendType.NOOP
+		_init_noop_backend()
+		return
+
+	print("VideoPlayer: Initializing AVPlayer backend for ", _source)
+
+	# Set initial state to loading
+	video_state = VIDEO_STATE_LOADING
+
+	# Create AVPlayer child node
+	var av_player_scene = load("res://src/decentraland_components/av_player.tscn")
+	av_player = av_player_scene.instantiate()
+	add_child(av_player)
+
+	# Wait for AVPlayer to be ready
+	await get_tree().process_frame
+
+	# Initialize texture with initial size (will be resized when video loads)
+	if not av_player.init_texture(640, 360):
+		push_error("VideoPlayer: Failed to initialize AVPlayer texture")
+		video_state = VIDEO_STATE_ERROR
+		return
+
+	# Set the video source
+	var success: bool
+	if _source.begins_with("http://") or _source.begins_with("https://"):
+		success = av_player.set_source_url(_source)
+	else:
+		# For local files, we need to fetch them first
+		await _async_fetch_and_set_local_source_av_player()
+		return
+
+	if not success:
+		push_error("VideoPlayer: Failed to set AVPlayer source: ", _source)
+		video_state = VIDEO_STATE_ERROR
+		return
+
+	av_player.set_looping(_is_looping)
+
+	if _is_playing:
+		av_player.play()
+
+
+func _init_noop_backend():
+	print("VideoPlayer: Using Noop backend (video playback not available)")
+	video_state = VIDEO_STATE_NONE
+	video_position = 0.0
+	video_length = -1.0
+
+
+func _process(_delta):
+	_update_effective_volume()
+	_update_video_state()
+
+
+## Calculate and apply effective volume for each backend
+## ExoPlayer/AVPlayer: effective_volume = master * scene * video_volume (bypasses Godot audio)
+## LiveKit: Only apply video_volume (Godot buses handle master/scene)
+func _update_effective_volume():
+	match current_backend:
+		BackendType.EXO_PLAYER:
+			_update_exo_player_volume()
+		BackendType.AV_PLAYER:
+			_update_av_player_volume()
+		BackendType.LIVEKIT:
+			_update_livekit_volume()
+		_:
+			pass
+
+
+## ExoPlayer bypasses Godot's audio system, so we calculate full effective volume
+func _update_exo_player_volume():
+	if not exo_player:
+		return
+
+	var effective_volume: float = 0.0
+
+	if not dcl_muted:
+		var config = Global.get_config()
+		var master_volume: float = config.audio_general_volume / 100.0
+		var scene_volume: float = config.audio_scene_volume / 100.0
+		effective_volume = master_volume * scene_volume * dcl_volume
+
+	if absf(effective_volume - _last_effective_volume) < 0.001:
+		return
+
+	_last_effective_volume = effective_volume
+	exo_player.set_volume(effective_volume)
+
+
+## AVPlayer bypasses Godot's audio system, so we calculate full effective volume
+func _update_av_player_volume():
+	if not av_player:
+		return
+
+	var effective_volume: float = 0.0
+
+	if not dcl_muted:
+		var config = Global.get_config()
+		var master_volume: float = config.audio_general_volume / 100.0
+		var scene_volume: float = config.audio_scene_volume / 100.0
+		effective_volume = master_volume * scene_volume * dcl_volume
```

Comment:
Almost identical functions, I think it is a better pattern to use something like

```
		BackendType.EXO_PLAYER:
			_update_mobile_player_volume(exo_player)
		BackendType.AV_PLAYER:
			_update_mobile_player_volume(av_player)
```